### PR TITLE
#435 Fix zero with scale to boolean (main)

### PR DESCRIPTION
**Pull request by stevenylai**
Fixes https://github.com/ezylang/EvalEx/issues/435 in the main branch

### CODE REVIEW

**Comment by Adler95** [comment]
@stevenylai I don't think this is the appropriate fix, as it doesn't really solve the: https://github.com/ezylang/EvalEx/issues/435.
The problem must be somewhere in the evaluation process since EvaluationValue already contains incorrect result.

**Comment by stevenylai** [comment]
It's a follow-up for https://github.com/ezylang/EvalEx/issues/417 where the fix is to replace `.equals()` with `.compareTo()`. We missed the int -> boolean conversion part from the previous fix.

**Comment by Adler95** [comment]
All good. I agree with you that there is a necessity to replace .equals() with .compareTo(). I just wanted to point out that it doesn't fix the issues referenced in your PR comment. https://github.com/ezylang/EvalEx/issues/435 :)

**Comment by Adler95** [comment]
@stevenylai I just did a bit more detailed review of the process (at first I thought that this method is not invoked during evaluation). I can see now that this is the proper fix. I apologize for the false information and thank you for providing the fix quickly. :)

File: src/main/java/com/ezylang/evalex/data/EvaluationValue.java [hunk 1]
```diff
@@ -374,7 +374,7 @@ public String getStringValue() {
   public Boolean getBooleanValue() {
     switch (getDataType()) {
       case NUMBER:
-        return !value.equals(BigDecimal.ZERO);
+        return ((BigDecimal) value).compareTo(BigDecimal.ZERO) != 0;
```
Comment (uklimaschewski) [inline src/main/java/com/ezylang/evalex/data/EvaluationValue.java:377]: I'd do it a little bit cleaner:
`return getNumberValue().compareTo(BigDecimal.ZERO) != 0;`

`getNumberValue()`already does the cast.

**Review by uklimaschewski — CHANGES_REQUESTED** [review]
Thanks, you were a little bit faster than me.
I have just one request: Can you not do the cast directly, but use the `getNumberValue()` method, instead?

**Review by uklimaschewski — APPROVED** [review]

**Pull request merged**